cmd/golangBuild.go

```diff
@@ -631,3 +634,19 @@ func isMainPackage(utils golangBuildUtils, pkg string) (bool, error) {
 
 	return true, nil
 }
+
+func gitConfigurationForPrivateModule(privateMod string, token string, utils golangBuildUtils) error {
+	privateMod = strings.ReplaceAll(privateMod, "/*", "")
+	modules := strings.Split(privateMod, ",")
+	for _, v := range modules {
+		authenticatedRepoURL := fmt.Sprintf("https://%s@%s", token, v)
```

Comment:
Could it occur that `v` already starts with `http(s)://`?

Reply:
I'm not pretty sure that what kind of url user will provide for us, but in our documentation we put the offical link of [GOPRIVATE](https://pkg.go.dev/cmd/go#hdr-Configuration_for_downloading_non_public_code). They should follow this rule